pkg/services/ngalert/notifier/alertmanager.go

```diff
@@ -144,23 +161,23 @@ func (am *Alertmanager) SyncAndApplyConfigFromDatabase() error {
 	am.reloadConfigMtx.Lock()
 	defer am.reloadConfigMtx.Unlock()
 
-	// TODO: check if config is same as before using hashes and skip reload in case they are same.
-	cfg, err := am.getConfigFromDatabase()
-	if err != nil {
-		return errors.Wrap(err, "get config from database")
-	}
-	return errors.Wrap(am.applyConfig(cfg), "reload from config")
-}
-
-func (am *Alertmanager) getConfigFromDatabase() (*api.PostableUserConfig, error) {
 	// First, let's get the configuration we need from the database.
 	q := &ngmodels.GetLatestAlertmanagerConfigurationQuery{}
 	if err := am.Store.GetLatestAlertmanagerConfiguration(q); err != nil {
-		return nil, err
+		// If there's no configuration in the database, let's use the default configuration.
+		if errors.Is(err, store.ErrNoAlertmanagerConfiguration) {
+			q.Result = &ngmodels.AlertConfiguration{AlertmanagerConfiguration: alertmanagerDefaultConfiguration}
+		} else {
+			return errors.Wrap(err, "unable to get Alertmanager configuration from the database")
+		}
 	}
 
-	// Then, let's parse and return the alertmanager configuration.
-	return Load(q.Result.AlertmanagerConfiguration)
+	cfg, err := Load([]byte(q.Result.AlertmanagerConfiguration))
+	if err != nil {
+		return err
+	}
+
+	return errors.Wrap(am.applyConfig(cfg), "unable to reload configuration")
```

Comment:
I think it would be better to use [am.ApplyConfig()](https://github.com/grafana/grafana/blob/258578766b43edd042c3d6e6ad4154ad796fe244/pkg/services/ngalert/notifier/alertmanager.go#L167) that it uses locking.

Reply:
The lock is already being held at the beginning of the function, we don't want two instance of this function to ever be looking at the database at the same time.

Reply:
👍  sorry I've totally overseen this